docs/syntax_and_semantics/offsetof.md

```diff
@@ -1,20 +1,27 @@
 # offsetof
 
-An `offsetof` expression returns the byte offset of an instance variable in a struct or class type. It accepts the type as first argument and the instance variable name prefixed by an `@` as second argument:
+An `offsetof` expression returns the byte offset of a field in an instance of a class or struct.
 
-```crystal
-offsetof(Type, @ivar)
-```
-
-This is a low-level primitive and only useful if a C API needs to directly interface with the data layout of a Crystal type.
-
-Example:
+There are two forms of `offsetof` expressions. The first form accepts any type as first argument and an instance variable name prefixed by an `@` as second argument, and returns the byte offset of that instance variable relative to an instance of the given type:
 
 ```crystal
 struct Foo
   @x = 0_i64
-  @y = 34_i32
+  @y = 34_i8
+  @z = 42_u16
 end
 
+offsetof(Foo, @x) # => 0
 offsetof(Foo, @y) # => 8
+offsetof(Foo, @z) # => 10
 ```
+
+The second form accepts any [`Tuple`](https://crystal-lang.org/api/latest/Tuple.html) instance type as first argument and an integer literal index as second argument, and returns the byte offset of the corresponding tuple element relative to an instance of the given type:
+
+```crystal
+offsetof(Tuple(Int64, Int8, UInt16), 0) # => 0
+offsetof(Tuple(Int64, Int8, UInt16), 1) # => 8
+offsetof(Tuple(Int64, Int8, UInt16), 2) # => 10
```

Comment:
This is very curious. If I had to guess, the initial intention probably was for this syntax to look like
`offsetof({Int64, Int8, UInt16}, 0)` - which *does* work.
Of course this is an example of the duality of the type grammar/syntax.

Whether you want to represent that in the example is up to you. Because writing it your way *is* actually clearer, now that I think about it.